libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/junction/api/MapboxJunctionApi.kt

```diff
@@ -0,0 +1,127 @@
+package com.mapbox.navigation.ui.maps.guidance.junction.api
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.common.HttpResponse
+import com.mapbox.common.core.module.CommonSingletonModuleProvider
+import com.mapbox.navigation.ui.base.model.Expected
+import com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionAction
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionProcessor
+import com.mapbox.navigation.ui.maps.guidance.junction.JunctionResult
+import com.mapbox.navigation.ui.maps.guidance.junction.model.JunctionError
+import com.mapbox.navigation.ui.maps.guidance.junction.model.JunctionValue
+import com.mapbox.navigation.ui.maps.guidance.junction.model.MapboxJunctionRequest
+import com.mapbox.navigation.utils.internal.JobControl
+import com.mapbox.navigation.utils.internal.ThreadController
+import kotlinx.coroutines.launch
+
+/**
+ * Mapbox Junction Api allows you to generate junction for select maneuvers.
+ * @property accessToken String
```

Comment:
Should we add a note here remarking that a special `accessToken` is needed for this feature to work?

Reply:
This will be handled in the docs as we did in `v1.x`

Reply:
Should we add a note to KDocs though too? 🤔